lib/oli_web/controllers/metrics_controller.ex

```diff
@@ -0,0 +1,26 @@
+defmodule OliWeb.MetricsController do
+  use OliWeb, :controller
+  require Logger
+
+  @doc """
+  Endpoint that triggers download of a container specific progress report.
+  """
+  def download_container_progress(conn, %{"section_slug" => slug, "container_id" => container_id} = params) do
+
+    case Oli.Delivery.Sections.get_section_by_slug(slug) do
+      nil ->
+        OliWeb.StaticPageController.not_found(conn, params)
+
+      section ->
+        contents = Oli.Delivery.Metrics.progress_datatable_for(section.id, container_id)
+        |> Oli.Analytics.DataTables.DataTable.to_csv_content()
+
+        conn
+        |> send_download({:binary, contents},
+          filename: "progress.csv"
```

Comment:
NIT: Perhaps we might want to append the section slug to the file name so it's easier to find?